src/formats/notion-api.ts

```diff
@@ -1934,57 +1981,79 @@ export class NotionAPIImporter extends FormatImporter {
 	}
 
 	/**
-	 * Check if a file should be skipped during import
-	 * This applies to BOTH incremental and full import modes
-	 * 
+	 * Check if a file should be skipped, updated, or created during import
+	 * Supports delta sync by comparing timestamps when incremental import is enabled
+	 *
 	 * @param filePath - Path to the file to check
 	 * @param notionId - Notion ID of the page being imported
+	 * @param pageLastEdited - Last edited timestamp from Notion API
 	 * @param ctx - Import context for reporting
-	 * @returns true if file should be skipped, false otherwise
+	 * @returns Object indicating action: 'skip', 'update', or 'create'
 	 */
-	private async shouldSkipExistingFile(
+	private async shouldSkipOrUpdateFile(
 		filePath: string,
 		notionId: string,
+		pageLastEdited: string,
 		ctx: ImportContext
-	): Promise<boolean> {
+	): Promise<{ action: 'skip' | 'update' | 'create' }> {
 		// Check if file exists
 		const file = this.vault.getAbstractFileByPath(normalizePath(filePath));
 		if (!file || !(file instanceof TFile)) {
-			return false; // File doesn't exist, don't skip
+			return { action: 'create' }; // File doesn't exist
 		}
 
 		// Read file and extract notion-id from frontmatter
 		try {
 			const content = await this.vault.read(file);
 			const notionIdMatch = content.match(/^notion-id:\s*(.+)$/m);
```

Comment:
While this function is being updated, it would be good to change this to read out of the metadataCache too. See the other related comment for an example.